packages/amplify-graphql-model-transformer/src/resources/model-resource-generator.ts

```diff
@@ -0,0 +1,670 @@
+import {
+  DataSourceProvider, MutationFieldType, QueryFieldType, SubscriptionFieldType,
+  TransformerContextProvider,
+  TransformerResolverProvider,
+} from '@aws-amplify/graphql-transformer-interfaces';
+import { ObjectTypeDefinitionNode } from 'graphql';
+import { MappingTemplate } from '@aws-amplify/graphql-transformer-core';
+import {
+  ResolverResourceIDs,
+  toCamelCase,
+} from 'graphql-transformer-common';
+import {
+  generateAuthExpressionForSandboxMode,
+  generateResolverKey,
+  ModelVTLGenerator,
+} from '../resolvers';
+import { ModelDirectiveConfiguration, SubscriptionLevel } from '../directive';
+import { ModelTransformerOptions } from '../types';
+
+/**
+ * Abstract class definition for ModelResourceGenerator
+ * ModelResourceGenerator implementations are intended to create resources for the model transformer plugin
+ * according to the data source that backs the model
+ */
+export abstract class ModelResourceGenerator {
+  protected datasourceMap: Record<string, DataSourceProvider> = {};
+  private resolverMap: Record<string, TransformerResolverProvider> = {};
+  protected generatorType = 'ModelResourceGenerator';
+  private enabled = false;
+  private provisioned = false;
+  private unprovisioned = false;
```

Comment:
What's the purpose of these flags?

Reply:
I put these in mostly because of looking at what was in the `before` method for the model transformer, we create parameters for managing the CFN values of the tables in DDB, but if someone only had RDS models, we wouldn't need to create those. Or if someone only had brownfield DDB, they wouldn't need those

I wanted a way to prevent creating resources that don't have any purpose